Source/WebCore/animation/AnimationEffect.cpp

```diff
@@ -110,30 +93,46 @@ AnimationEffectTiming::ResolutionData AnimationEffect::resolutionData(std::optio
     };
 }
 
-BasicEffectTiming AnimationEffect::getBasicTiming(std::optional<WebAnimationTime> startTime) const
+BasicEffectTiming AnimationEffect::getBasicTiming(std::optional<WebAnimationTime> startTime)
 {
+    updateComputedTimingPropertiesIfNeeded();
     return m_timing.getBasicTiming(resolutionData(startTime));
 }
 
-ComputedEffectTiming AnimationEffect::getBindingsComputedTiming() const
+ComputedEffectTiming AnimationEffect::getBindingsComputedTiming()
 {
     if (auto* styleOriginatedAnimation = dynamicDowncast<StyleOriginatedAnimation>(animation()))
         styleOriginatedAnimation->flushPendingStyleChanges();
     return getComputedTiming();
 }
 
-ComputedEffectTiming AnimationEffect::getComputedTiming(std::optional<WebAnimationTime> startTime) const
+ComputedEffectTiming AnimationEffect::getComputedTiming(std::optional<WebAnimationTime> startTime)
 {
+    updateComputedTimingPropertiesIfNeeded();
+
     auto data = resolutionData(startTime);
     auto resolvedTiming = m_timing.resolve(data);
 
+    // https://drafts.csswg.org/web-animations-2/#dom-animationeffect-getcomputedtiming
+    // The description of the duration attribute of the object needs to indicate that if timing.duration
+    // is the string auto, this attribute will return the current calculated value of the intrinsic iteration
+    // duration, which may be a expressed as a double representing the duration in milliseconds or a percentage
+    // when the effect is associated with a progress-based timeline.
+    auto computedDuration = [&]() -> DoubleOrCSSNumericValueOrString {
```

Comment:
What's the point of making this a function? You only call it once with no arguments, so call it here?

Reply:
I mean, making the variable a function, rather than just the value.

Reply:
No point, I'll make the suggested change.